src/structs.py

```diff
@@ -1133,10 +1133,14 @@ def __post_init__(self) -> None:
 
 @dataclass(frozen=True, eq=False, repr=False)
 class Query(abc.ABC):
-    """Base class for a Query.
+    """Base class for a Query."""
 
-    Has no API.
-    """
+    def __len__(self) -> int:
+        """The number of queries in this Query.
```

Comment:
we should def do what tom suggested instead of this, since it doesn't make much sense for an arbitrary query to have a thing called length